pkg/controller/secrets/tenant_controller.go

```diff
@@ -90,6 +90,17 @@ func AddTenantController(mgr manager.Manager, opts options.AddOptions) error {
 		return fmt.Errorf("tenant-controller failed to watch tenant CA Secret %s in all namespace: %w", certificatemanagement.TenantCASecretName, err)
 	}
 
+	// Make a helper for determining which namespaces to use based on tenancy mode.
+	// In multi-tenant mode, we need to watch all namespaces for secrets. In single-tenant mode,
+	// we only need to watch the elasticsearch namespace. Both need tigera-operator.
```

Comment:
This comment doesn't seem to align with the use of the `NamespaceHelper`, unless I've misunderstood what's going on. You're setting both single-tenant and multi-tenant namespaces to `""` so regardless of whether `opts.MultiTenant` is true or false you will use namespace `""` in the below watches. https://github.com/tigera/operator/blob/master/pkg/controller/utils/namespace_helper.go#L69